### DISCUSSION TITLE
Budgets - information vs data

### DISCUSSION THREAD

**Discussion by skorvek** [firefly-iii/firefly-iii#6205] [Firefly III]
How does one determine the average amount spent per budget over time?

I've opened individual budgets and get a listing of all component transactions and some pie charts.
On the far right, there's that month's expenditures, and if I click "show everything" I do NOT get a monthly (or per period as per the settings) breakdown of the amount spend like I see on the analogous categories pages.
I've found (somewhat not intuitive, but I found it) that I can pick an inclusive date range at the top of the page- but that right hand column always just shows what I spend last month. (Currently it's 7/1/2022, and the only box in that column is listing June 2022 budget total, despite setting the top of the page to April- June 2022).
 
As a matter of fact- manipulating the date range at the top of the page doesn't seem to do anything- the graph shows a weekly bar graph of all of my data regardless of date, and the transaction list shows everything and the monthly breakdown is last month.  Is this a bug? 

The category page seems to work as expected.

I've run the budget report, and averages are by asset account (not helpful to me in my setup) and by destination account (somewhat helpful) or by month. But by month they're only on the graph and unlabeled (have to hover) without a median or average line. It's not easy to grasp the average amount and the variability month to month from that graph- unlike the similar graph on accompanying an individual bill. That graph is great. It would be nice if there was a tabular rendition as well  as that would help input realistic budget numbers in the monthly budget.   Not sure you could do that easily on the budget page itself. On the report it seems very achievable though.

Same could be said for breaking down the budget by category. Pie charts are usually not too useful. Putting that data in tabular form and doing some simple stats (monthly average, category total over period) would be really useful information.

**Comment by JC5** [comment]
The default report may have the info you need. Set it to yearly and it will group by month.
